tests/test_datacube.py

```diff
@@ -0,0 +1,30 @@
+import sys
+import pytest
+
+sys.path.append('../modules/')
+from datacube import read_data, check_keys
+
+
+@pytest.mark.parametrize('actual_keys,required_keys,passed', [
+    ({'rlat', 'rlon', 'dv', 4, 'lat', 'lon'}, {'rlat', 'rlon', 'dv', 4}, True),
+    ({'rlat', 'rlon', True, 99999}, {'rlat', 'rlon', True, 99999}, True)])
+def test_check_keys(actual_keys, required_keys, passed):
+    checker = check_keys(actual_keys, required_keys)
+    assert(checker == passed)
+
+@pytest.mark.parametrize('actual_keys,required_keys,passed', [
+    ({'rlat', 'rlon', 4.0, False}, {'hi', 'nic', 'was', 'here', 'lon'}, False)])
+def test_bad_keys(actual_keys, required_keys, passed):
+    with pytest.raises(KeyError):
+        checker = check_keys(actual_keys, required_keys)
```

Comment:
Cleaner to merge these two sets of tests into one:

```
def test_check_keys(actual_keys, required_keys, passed):
    if passed:
        assert check_keys(actual_keys, required_keys)
    else:
         with pytest.raises(KeyError):
             check_keys(actual_keys, required_keys)
```

Reply:
I agree this code looks better, but @nikola-rados suggested that keeping the fail cases and the pass cases as separate as possible is often the best way forward - meaning separate functions. To implement your suggestion would mean I would combine all of the passes and fail parameters in one decorator. What are your thoughts @nikola-rados and @rod-glover? Does it matter?

Reply:
Maybe @jameshiebert has an opinion on this, too.

Reply:
I'd be interested to hear why @nikola-rados thinks so. It's not my opinion. Ultimately it doesn't matter much, but it is a minor violation of the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) rule.

Reply:
I'll leave you all to discuss among yourselves. I'm happy with whatever you collaboratively decide.

Reply:
I suggest we not spend too much time on this. It's a minor matter, and worth discussing for what can be learned, but it's not a very big deal either conceptually or pragmatically.

Reply:
Thanks guys. I'm going to lean in favour of Rod's suggested change since I am testing a function, `check_keys` that determines if the input is good or bad to begin with.